pyicloud/base.py

```diff
@@ -98,6 +121,14 @@ def authenticate(self):
             msg = 'Invalid email/password combination.'
             raise PyiCloudFailedLoginException(msg)
 
+        # Pull X-APPLE-WEB-KB cookie from cookies
+        NewWebKBCookie = next(({key:val} for key, val in req.cookies.items() if 'X-APPLE-WEB-KB' in key), None)
```

Comment:
You don't need to apologize for that kind of thing, @FiZiX -- the important part is that you're solving a problem, not how pretty it is!  

Awesome, thanks for the explanation; I think I have an idea of how to clean it up just a little.